compiler/rustc_trait_selection/src/traits/query/dropck_outlives.rs

```diff
@@ -110,7 +110,12 @@ pub fn trivial_dropck_outlives<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>) -> bool {
         // check if *any* of those are trivial.
         ty::Tuple(ref tys) => tys.iter().all(|t| trivial_dropck_outlives(tcx, t.expect_ty())),
         ty::Closure(_, ref substs) => {
-            substs.as_closure().upvar_tys().all(|t| trivial_dropck_outlives(tcx, t))
+            if !substs.as_closure().is_valid() {
```

Comment:
This can just be `trivial_dropck_outlives(tcx, substs.as_closure().tupled_upvar_tys())`